File: frontend/src/app/types/dto/person/related/contact/phone/person-phone-number.dto.ts
import { PhoneNumberDto } from '../../../../phone-number.dto';
import { PersonPhoneNumberDtoId } from '../../reference/contact/person-phone-number-dto-id';


export interface PersonPhoneNumberDto {

    id?: PersonPhoneNumberDtoId;

    phoneNumber: PhoneNumberDto;

    mainNumber: boolean;

}

File: backend/src/main/java/ibee/webapp/todo_app/core/dto/person/contact/phone/PersonPhoneNumberDto.java
package ibee.webapp.todo_app.core.dto.person.contact.phone;

import ibee.webapp.todo_app.core.dto.PhoneNumberDto;
import ibee.webapp.todo_app.features.person.related.referenceIds.contact.PersonPhoneNumberDtoId;
import jakarta.validation.constraints.NotNull;

public record PersonPhoneNumberDto(
    @NotNull
    PersonPhoneNumberDtoId id,

    @NotNull
    PhoneNumberDto phoneNumber,

    @NotNull
    Boolean mainPhoneNumber
) {

}
